### PR TITLE
Add Binder spackage

### PR DESCRIPTION
**Pull request by kliegeois**
This PR adds a spack recipe for the Binder software: https://github.com/RosettaCommons/binder

Related issue https://github.com/RosettaCommons/binder/issues/243

### CODE REVIEW

File: var/spack/repos/builtin/packages/binder/package.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    That is, Binder, takes a C++ project and compiles it into objects
+    and functions that are all usable within Python.
+    Binder is different from prior tools in that it handles special
+    features new in C++11.
+    """
+
+    homepage = "https://github.com/RosettaCommons/binder"
+    git = "https://github.com/RosettaCommons/binder.git"
+
+    maintainers("lyskov", "kliegeois")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/binder/package.py:22]: @lyskov Are you okay with being a maintainer on this package?  (See https://spack.readthedocs.io/en/latest/packaging_guide.html#maintainers.)

**Review by tldahlgren — APPROVED** [review]
Confirmed the homepage, branch, and tags.  Just waiting to get maintainer confirmation from @lyskov

**Comment by kliegeois** [comment]
Thanks @tldahlgren for the review and the merge!

File: var/spack/repos/builtin/packages/binder/package.py (hunk 1 above)
Comment (lyskov) [reply var/spack/repos/builtin/packages/binder/package.py:22]: @tldahlgren this is a question about maintainer for spack package? If so then no, i do not think i should be listed as maintainer for it. Thanks,

**Pull request merged**